include/pybind11/numpy.h

```diff
@@ -493,18 +492,13 @@ class array : public buffer {
                           " (ndim = " + std::to_string(ndim()) + ")");
     }
 
-    template<typename... Ix> size_t get_byte_offset(Ix&&... index) const {
-        const size_t idx[] = { (size_t) index... };
-        if (!std::equal(idx + 0, idx + sizeof...(index), shape(), std::less<size_t>{})) {
-            auto mismatch = std::mismatch(idx + 0, idx + sizeof...(index), shape(), std::less<size_t>{});
-            throw index_error(std::string("index ") + std::to_string(*mismatch.first) +
-                              " is out of bounds for axis " + std::to_string(mismatch.first - idx) +
-                              " with size " + std::to_string(*mismatch.second));
-        }
-        return std::inner_product(idx + 0, idx + sizeof...(index), strides(), (size_t) 0);
+    size_t byte_offset() const {
+        return 0;
     }
 
-    size_t get_byte_offset() const { return 0; }
```

Comment:
I guess I'd still prefer to have a separate `byte_offset_unchecked()` which would be used by all other methods (and xtensor), and having all other methods (including `byte_offset()`) be checked, so it's 'safe by default'.